File: Data Structures & Algorithms/string-encode-and-decode/submission-0.py
class Solution:

    def encode(self, strs):
        res = ""
        for s in strs:
            res += str(len(s)) + "$" + s
        return res

    def decode(self, str):
        res = []
        i = 0
        while i < len(str):
            j = i
            while str[j] != "$":
                j += 1
            size = int(str[i:j])
            word = str[j + 1 : j + 1 + size]
            res.append(word)
            i = j + 1 + size
        return res

File: Data Structures & Algorithms/climbing-stairs/submission-0.java
class Solution {
    public int climbStairs(int n) {
        int[] dp = new int[n+1];
        if(n<= 2){
            return n;
        }
        dp[n] = 1;
        dp[n-1] = 1;
        for(int j = n-2; j>=0 ; j--){
            dp[j] = dp[j+1]+dp[j+2];
        }
        return dp[0];
    }
}
